python/lib/communication/dmod/communication/scheduler_request.py

```diff
@@ -40,21 +45,22 @@ def factory_init_from_deserialized_json(cls, json_obj: dict):
         try:
             model_request = ModelExecRequest.factory_init_correct_subtype_from_deserialized_json(json_obj['model_request'])
             if model_request is not None:
+                alloc_paradigm = json_obj['allocation_paradigm'] if 'allocation_paradigm' in json_obj else None
```

Comment:
Not enough to derail the PR, but the `'allocation_paradigm'` magic string might need to be defined in a more concrete fashion elsewhere (maybe at the top?). This is probably inconsistent in that I probably haven't noticed this elsewhere and I definitely do it myself. Defining it elsewhere isn't going to prevent the direct usage, but it could help goobers (i.e. me) from typing it in wrong or using a different term elsewhere, like naming it 'allocationparadigm` or something and only finding the error later.

Reply:
Yeah, I have defined serialization keys as "private" class variables with some-but-not-all of the serializable types.  I agree:  at some point we should go back and refactor that in places it isn't already done.